pkg/controller/workload/workload_processor.go

```diff
@@ -318,45 +314,39 @@ func (p *Processor) storeServiceCache(workload_uid string, serviceName string, p
 	endpointCaches[workload_uid] = endpoint
 }
 
-func (p *Processor) storeBackendData(uid uint32, ips [][]byte, portList *workloadapi.PortList, waypoint *workloadapi.GatewayAddress, serviceName string) error {
+func (p *Processor) storeBackendData(uid uint32, ip []byte, waypoint *workloadapi.GatewayAddress, portList map[string]*workloadapi.PortList) error {
 	var (
-		err error
-		bk  = bpf.BackendKey{}
-		bv  = bpf.BackendValue{}
+		bk = bpf.BackendKey{}
+		bv = bpf.BackendValue{}
 	)
 
 	bk.BackendUid = uid
-	for _, ip := range ips {
-		if waypoint != nil {
-			addr := waypoint.GetAddress().Address
-			bv.WaypointAddr = nets.ConvertIpByteToUint32(addr)
-			bv.WaypointPort = nets.ConvertPortToBigEndian(waypoint.GetHboneMtlsPort())
+	bv.IPv4 = nets.ConvertIpByteToUint32(ip)
+	bv.ServiceCount = 0
+	for serviceName := range portList {
+		bv.Services[bv.ServiceCount] = p.hashName.StrToNum(serviceName)
```

Comment:
In the code, some Map such as `endpointsByService` using serviceName(string key) for store, however， call StrToNum when map is read. If the map is directly stored by number, conversion is not required and more efficiency.

Reply:
This is mainly about the optimization related to endpointsByService, which may be carried out later and is not related to the current PR processing problem.